lib/src/sip_ua_helper.dart

```diff
@@ -78,40 +101,42 @@ class SIPUAHelper extends EventManager {
       this._ua = UA(_settings);
       this._ua.on(EventConnecting(), (EventConnecting event) {
         logger.debug('connecting => ' + event.toString());
-        _handleSocketState('connecting', null);
+        _notifyRegsistrationStateListeners(RegistrationStateEnum.CONNECTING);
       });
 
       this._ua.on(EventConnected(), (EventConnected event) {
         logger.debug('connected => ' + event.toString());
-        _handleSocketState('connected', null);
+        _notifyRegsistrationStateListeners(RegistrationStateEnum.CONNECTED);
         _connected = true;
       });
 
       this._ua.on(EventDisconnected(), (EventDisconnected event) {
         logger.debug('disconnected => ' + event.toString());
-        _handleSocketState('disconnected', null);
+        _notifyRegsistrationStateListeners(RegistrationStateEnum.DISCONNECTED);
         _connected = false;
       });
 
       this._ua.on(EventRegistered(), (EventRegistered event) {
         logger.debug('registered => ' + event.toString());
         _registered = true;
-        _registerState = 'registered';
-        _handleRegisterState('registered', event.response);
+        _registerState = RegistrationStateEnum.REGISTERED;
+        _notifyRegsistrationStateListeners(RegistrationStateEnum.REGISTERED);
       });
 
       this._ua.on(EventUnregister(), (EventUnregister event) {
         logger.debug('unregistered => ' + event.toString());
-        _registerState = 'unregistered';
+        _registerState = RegistrationStateEnum.UNREGISTERED;
         _registered = false;
-        _handleRegisterState('unregistered', event.response);
+        _notifyRegsistrationStateListeners(RegistrationStateEnum.UNREGISTERED);
       });
 
       this._ua.on(EventRegistrationFailed(), (EventRegistrationFailed event) {
         logger.debug('registrationFailed => ' + (event.cause));
-        _registerState = 'registrationFailed[${event.cause}]';
+        _registerState = RegistrationStateEnum
```

Comment:
Need a way to send `code` and `cause` to the UI, telling why the registration failed.

Reply:
code is not available, and as only cause is available I'll just add an extra parameter to the listener like this...

 void registrationStateChanged(RegistrationStateEnum state, String cause);